cmake/add_module.cmake

```diff
@@ -9,12 +9,15 @@ include(isolate_headers)
 # that cannot include headers from other directories in include/
 # unless they come through linked libraries.
 #
-# add_module(parent a)
-# add_module(parent b)
+# add_module(parent a true)
+# add_module(parent b false)
 # target_link_libraries(project.libparent.b PUBLIC project.libparent.a)
-function(add_module parent name)
+function(add_module parent name unity)
   set(target ${PROJECT_NAME}.lib${parent}.${name})
   add_library(${target} OBJECT)
```

Comment:
`add_library` will respect the value of [`CMAKE_UNITY_BUILD`](https://cmake.org/cmake/help/latest/variable/CMAKE_UNITY_BUILD.html#variable:CMAKE_UNITY_BUILD) when it is called. The `UNITY_BUILD` property of a new target takes its default value from `CMAKE_UNITY_BUILD`. I think you should just set `CMAKE_UNITY_BUILD` before the calls to `add_module` instead of modifying the function.

Reply:
You make a good point, but to do it right we probably ought to follow the advice in the link you provided, and deprecate `option(unity ...)` in favour of reading (but not setting) `CMAKE_UNITY_BUILD`, and then switch our CI to set `CMAKE_UNITY_BUILD`

Reply:
alternatively we ignore the advice `Projects should not set this variable` and just set it very early in `CMakeLists.txt` I guess ??

Reply:
I think because we're setting `CMAKE_UNITY_BUILD` based on user input (`unity`), we're morally in the clear. I do think that getting rid of the option is a good idea, but beyond the appropriate scope of this PR, which is just intended as a bug fix.

Reply:
@thejohnfreeman @Bronek Once again, I reverted the commit with my original change, and added a new commit with the simpler solution John suggested. Again it appears to be working perfectly. I'll probably run through all the CI builds tomorrow to check they're following the `unity` setting as expected.